mobile/library/common/internal_engine.cc

```diff
@@ -156,6 +156,9 @@ envoy_status_t InternalEngine::main(std::shared_ptr<Envoy::OptionsImplBase> opti
         Envoy::Server::ServerLifecycleNotifier::Stage::PostInit, [this]() -> void {
           ASSERT(Thread::MainThread::isMainOrTestThread());
 
+          // StrippedMainBase sets the `new` handler but in Envoy Mobile, the application using it
+          // should control malloc failure behavior, so it's set to nullptr.
+          std::set_new_handler(nullptr);
```

Comment:
drive-by: Is it possible that the application has already set their handler by the time this code runs? if so, that would result in the hander going from:

1. nullptr -> app handler (in the app)
2. app handler -> envoy handler (in StrippedMainBase)
3. envoy handler -> nullptr (here)

I'm not sure if that's possible, but if so it might make sense to instead have some sort of config/api/whatever in core envoy to disable setting the handler.

Reply:
That's a great point!  Updated with a configuration option, WDYT?